### PR TITLE
Re-implement thesaurus import/export using rdflib to support a wider range of RDF formats [Fixes #8701]

### LINKED ISSUE #8701: Cannot upload valid SKOS thesauri (plus other issues with thesaurus import)
## Expected Behavior

To be able to create a thesaurus via upload of a valid SKOS RDF file (RDF/XML or other formats like TTL or JSON-LD).

## Actual Behavior

Upload succeeds for simple thesauri in "pretty XML"-favoured RDF, e.g. files that look like:

```xml
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <skos:ConceptScheme xmlns:skos="http://www.w3.org/2004/02/skos/core#" rdf:about="http://example.com/thesaurus">
    <dc:title xmlns:dc="http://purl.org/dc/elements/1.1/" xml:lang="en">Example</dc:title>
    ....
  </skos:ConceptScheme>
</rdf:RDF>

```

However, it's pretty common to serialise RDF/XML like this:

```xml
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"> 
  <rdf:Description rdf:about="http://example.com/thesaurus">
    <rdf:type rdf:resource="http://www.w3.org/2004/02/skos/core#ConceptScheme"/>
    <dc:title xmlns:dc="http://purl.org/dc/elements/1.1/" xml:lang="en">Example</dc:title>
    ....
  </rdf:Description>
</rdf:RDF>
```

The `load_thesaurus` command only supports the first version: uploading the other XML serialisation format results in an error: "ConceptScheme not found in file".

Other common RDF formats like Turtle and JSON-LD are not currently supported.

## Steps to Reproduce the Problem

  1. Go to `/en/admin/base/thesaurus/`
  2. Click "Upload Thesaurus"
  3. Select a valid SKOS-format RDF file in anything other than the first XML format described above

## Specifications

  - GeoNode version: 3.2.1 and master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker):
  - Platform: Ubuntu 20.04
  - Additional details:

The root of the problem here is that the `load_thesaurus` command is trying to read the data as XML rather than RDF, but RDF has some nuances and a variety of flavours.

There are some other, more minor, issues with thesaurus import:

 - the varchar field for language tags is 3 chars long, but it's not uncommon to find language tags like `"foobar"@ru-Latn` or `"foobar"@ru-Cyrl`, i.e. with the script code qualifier, or other qualifiers (there's no real length limit AFAICR.)
 - the `altLabel` field is used as a sort of language-agnostic fallback value, but it's pretty normal that altLabels would be multilingual as well

I will put in a PR that addresses some of these issues by:

 - using `rdflib` to parse the RDF, which should handle all flavours of RDF/XML as well as other serialisation formats
 - increasing the length of the keyword label field

I won't change the existing behaviour of the `altLabel`s since I assume people rely on it working the way it does now.

### PR DESCRIPTION
**Pull request by mikesname**
This patch revises the `load_thesaurus` and `dump_thesaurus` management commands to use [`rdflib`](https://rdflib.readthedocs.io/en/stable/), rather than bespoke XML parsing. This allows importing and exporting a range of RDF formats rather than just one flavour of RDF/XML.

Aside from these commands and the addition of `rdflib` to the `requirements.txt`, the only other change is increasing the length of the `ThesaurusKeywordLabel.lang` field from 3 to 255, since there's not really a limit on RDF language tag length.

I have added additional tests for the `dump_thesaurus` command.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [x] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@ Deprecated==1.2.13
 wrapt==1.13.3
 jsonschema==4.3.3
 zipstream-new==1.1.8
+rdflib==6.1.1
```
[the remark this answers is not in the document]
Comment (mikesname) [reply requirements.txt:25]: Done!

**Comment by mikesname** [comment]
Sorry, I forced pushed a couple of changes here when I had to redo the migration with updates from master. Shalln't do that again. If need be I'll convert the PR to a draft till I've worked through the inevitable issues...

**Review by mattiagiupponi — CHANGES_REQUESTED** [review]
Hi, @mikesname thanks for the contribution.
Overall looks good, just a few points:
- Can u also update the [official documentation](https://github.com/GeoNode/documentation) for the `dump_thesaurus` command? So we can keep it updated with the new parameters that are added to the command
- I have some doubts related to the output format of the dump.
I usually use the "GEMET - INSPIRE themes, version 1.0" thesaurus for my test, and the beginning of the file looks wrong. 
Original file
```xml
<skos:ConceptScheme xmlns:skos="http://www.w3.org/2004/02/skos/core#" rdf:about="http://inspire.ec.europa.eu/theme">
  <dc:title xmlns:dc="http://purl.org/dc/elements/1.1/">GEMET - INSPIRE themes, version 1.0</dc:title>
  <dcterms:issued xmlns:dcterms="http://purl.org/dc/terms/">2018-05-23T10:25:56</dcterms:issued>
  <dcterms:modified xmlns:dcterms="http://purl.org/dc/terms/">2018-05-23T10:25:56</dcterms:modified>
</skos:ConceptScheme>
<skos:Concept xmlns:skos="http://www.w3.org/2004/02/skos/core#" rdf:about="http://inspire.ec.europa.eu/theme/of">
  <skos:prefLabel xml:lang="en">Oceanographic geographical features</skos:prefLabel>
  <skos:scopeNote xml:lang="en">Physical conditions of oceans (currents, salinity, wave heights, etc.).</skos:scopeNote>
  <skos:altLabel>of</skos:altLabel>
  <skos:inScheme rdf:resource="http://inspire.ec.europa.eu/theme" />
</skos:Concept>
```

Dump:
```xml
<skos:Concept rdf:about="http://inspire.ec.europa.eu/theme/of">
    <skos:inScheme>
      <skos:ConceptScheme rdf:about="http://inspire.ec.europa.eu/theme">
        <dc:title>GEMET - INSPIRE themes, version 1.0</dc:title>
        <dc:description>GEMET - INSPIRE themes, version 1.0</dc:description>
        <dcterms:issued>2018-05-23T10:25:56</dcterms:issued>
      </skos:ConceptScheme>
    </skos:inScheme>
    <skos:altLabel>of</skos:altLabel>
    <skos:prefLabel xml:lang="en">Oceanographic geographical features</skos:prefLabel>
  </skos:Concept>
```
It looks like the header of the thesaurus has been mixed with one of the labels.

Following the file used:
- inspire-theme (the original thesaurus)
- test2 the output from the dump
[thesaurus.zip](https://github.com/GeoNode/geonode/files/7969552/thesaurus.zip)

(had to zip them since GitHub doesn't support RDF

**Comment by mikesname** [comment]
Hi @mattiagiupponi, okay I've attempted to put in a [PR for that](https://github.com/GeoNode/documentation/pull/217).

**Pull request merged**